File: c/58.c
#include<stdio.h> 


int lengthOfLastWord(char * s){
    int i;
    int size;
    int end;
    size = 0;
    end = 0;
    i = 0;
    while(s[i] != '\0'){
        if(s[i] != ' '){
            if(end == 1){
                size = 0;
                end = 0;
            }
            size++;
        }else{
            end = 1;
        }
        i++;
    }
    return size;
}

int main(){
    int len;
    char* s;
    s = "";
    len = lengthOfLastWord(s);
    printf("%d\n", len);
}

File: python/58.py
class Solution(object):
    # Solution 1
    # def lengthOfLastWord(self, s):
    #     """
    #     :type s: str
    #     :rtype: int
    #     """
    #     if not s:
    #         return 0
    #     words = s.split(' ')
    #     words = filter(lambda x:x, words)
    #     if not words:
    #         return 0
    #     return len(words[-1])

    # Solution 2 - Faster
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        cnt = 0
        for c in s[::-1]:
            if c != ' ':
                cnt += 1
            if cnt != 0 and c == ' ':
                return cnt
        return cnt

if __name__ == "__main__":
    a = "  a   "
    sol = Solution()
    print(sol.lengthOfLastWord(a))